### PR TITLE
rivercarro: 0.1.4 -> 0.2.1

### PR DESCRIPTION
**Pull request by adamcstephens**
###### Description of changes
https://git.sr.ht/~novakane/rivercarro/refs/v0.2.0
https://git.sr.ht/~novakane/rivercarro/refs/v0.2.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by adamcstephens** [comment]
Bumping to keep in line with river.

File: pkgs/applications/misc/rivercarro/default.nix [hunk 1]
```diff
@@ -4,25 +4,27 @@
...
-    rev = "v${version}";
-    sha256 = "sha256-eATbbwIt5ytEVLPodyq9vFF9Rs5S1xShpvNYQnfwdV4=";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-lI1xU7Qw4+XmFLnFxVZvrAPMZs0SNStbcUPBqSYUBak=";
   };
 
   nativeBuildInputs = [
     pkg-config
     river
     wayland
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rivercarro/default.nix:26]: wayland goes to buildInputs

Comment (adamcstephens) [reply pkgs/applications/misc/rivercarro/default.nix:25]: moved wayland and river to buildInputs.

File: pkgs/applications/misc/rivercarro/default.nix [hunk 2]
```diff
@@ -4,25 +4,30 @@
...
+  buildInputs = [
     river
     wayland
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    wayland-protocols
     zig
   ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rivercarro/default.nix:32]:
```suggestion
  nativeBuildInputs = [
    pkg-config
    wayland-protocols
    zig
  ];

  buildInputs = [
    river
    wayland
  ];
```

File: pkgs/applications/misc/rivercarro/default.nix [hunk 3]
```diff
@@ -46,4 +51,3 @@ stdenv.mkDerivation rec {
     maintainers = with maintainers; [ kraem ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rivercarro/default.nix:51]: not sure what is wrong with aarch64-linux but I would suggest to mark it broken on aarch64-linux

```
LLVM Emit Object... Semantic Analysis [2361] gfx1036... Semantic Analysis [2719] btver2... Semantic Analysis [2977] getOrPut... Semantic Analysis [3263] reallocAtLeast... Semantic Analysis [3347] eventfd... Semantic Analysis [3476] sizes... Semantic Analysis [3609] isEmpty... Semantic Analysis [3722] exeFileExtSimple... Semantic Analysis [3828] count... Semantic Analysis [3948] debug... Semantic Analysis [4007] iterator... Semantic Analysis [4110] grow... Semantic Analysis [4179] minimal_capacity... Semantic Analysis [4232] allocatedSlice... Semantic Analysis [4256] minimal_capacity... LLVM Emit Object... LLVM Emit Object... Semantic Analysis [4266] LLVM Emit Object... LLVM Emit Object... LLD Link... LLD Link... LLVM Emit Object... LLVM Emit Object... Semantic Analysis [1613] /build/source/zig-cache/zig-wayland/common.zig:103:21: error: cast increases pointer alignment
                    @ptrCast(*const fn (*Obj, Payload, Data) void, implementation)(
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/zig-cache/zig-wayland/common.zig:103:68: note: '?*const anyopaque' has alignment '1'
                    @ptrCast(*const fn (*Obj, Payload, Data) void, implementation)(
                                                                   ^~~~~~~~~~~~~~
/build/source/zig-cache/zig-wayland/common.zig:103:30: note: '*const fn(*wayland_client.Registry, wayland_client.Registry.Event, *main.Context) void' has alignment '4'
                    @ptrCast(*const fn (*Obj, Payload, Data) void, implementation)(
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
referenced by:
    setListener: /build/source/zig-cache/zig-wayland/wayland_client.zig:159:56
    main: /build/source/src/main.zig:450:13
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
/build/source/zig-cache/zig-wayland/common.zig:103:21: error: cast increases pointer alignment
                    @ptrCast(*const fn (*Obj, Payload, Data) void, implementation)(
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/zig-cache/zig-wayland/common.zig:103:68: note: '?*const anyopaque' has alignment '1'
                    @ptrCast(*const fn (*Obj, Payload, Data) void, implementation)(
                                                                   ^~~~~~~~~~~~~~
/build/source/zig-cache/zig-wayland/common.zig:103:30: note: '*const fn(*river_layout_v3_client.LayoutV3, river_layout_v3_client.LayoutV3.Event, *main.Output) void' has alignment '4'
                    @ptrCast(*const fn (*Obj, Payload, Data) void, implementation)(
                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: rivercarro...
error: The following command exited with error code 1:
```

Comment (adamcstephens) [reply pkgs/applications/misc/rivercarro/default.nix:52]: Yeah, I'm not sure why it's failing now on aarch64-linux either. I've marked it as broken and will let someone who actually uses this package decide if they care to fix it.

File: pkgs/applications/misc/rivercarro/default.nix [hunk 4]
```diff
@@ -43,7 +48,7 @@ stdenv.mkDerivation rec {
     description = "A layout generator for river Wayland compositor, fork of rivertile";
     license = licenses.gpl3Plus;
     platforms = platforms.linux;
+    broken = stdenv.isAarch64;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rivercarro/default.nix:51]: This should have a little comment why the package was marked broken

**Review by SuperSandro2000 — COMMENTED** [review]
I think we can squash all the commits

**Pull request closed without merging**